### PR TITLE
DEV: Remove summarization code

### PR DESCRIPTION
**Pull request by keegangeorge**
This PR removes any code related to summarization. Any logic for handling summarization will now return to being handled by the `discourse-ai` plugin. See PR: https://github.com/discourse/discourse-ai/pull/658 for the logic being added back to `discourse-ai`.

### CODE REVIEW

File: config/site_settings.yml [hunk 1]
```diff
@@ -2644,16 +2644,6 @@ uncategorized:
     client: true
     default: false
 
-  summarization_strategy:
```
Comment (romanrizzi) [inline config/site_settings.yml:2647]: I believe we want to keep these hidden so we can migrate data. We can drop them later.

File: config/site_settings.yml [hunk 2]
```diff
@@ -2644,16 +2644,6 @@ uncategorized:
...
-    client: true
-    default: ""
-    enum: "SummarizationStrategy"
-    validator: "SummarizationValidator"
-  custom_summarization_allowed_groups:
```
Comment (romanrizzi) [inline config/site_settings.yml:2652]: I believe we want to keep these hidden so we can migrate data. We can drop them later.

File: plugins/chat/app/controllers/chat/api/summaries_controller.rb [hunk 3]
```diff
@@ -10,9 +10,11 @@ def get_summary
     channel = Chat::Channel.find(params[:channel_id])
     guardian.ensure_can_join_chat_channel!(channel)
 
-    strategy = Summarization::Base.selected_strategy
+    strategy = DiscourseAi::Summarization::Models::Base.selected_strategy
```
Comment (romanrizzi) [inline plugins/chat/app/controllers/chat/api/summaries_controller.rb:13]: Chat summaries now depend on a plugin that it's not always installed.

**Review by romanrizzi — CHANGES_REQUESTED** [review]

File: plugins/chat/app/controllers/chat/api/summaries_controller.rb (hunk 3 above)
Comment (xfalcox) [reply plugins/chat/app/controllers/chat/api/summaries_controller.rb:13]: Indeed, chat must follow the same idea of having it's AI summarization bits extracted and moved into the plugin or simply deleted.

File: app/assets/javascripts/discourse/app/components/post-text-selection.gjs [hunk 4]
```diff
@@ -226,6 +226,7 @@ export default class PostTextSelection extends Component {
       offset,
       trapTab: false,
       closeOnScroll: false,
+      modalForMobile: false,
```
Comment (romanrizzi) [inline app/assets/javascripts/discourse/app/components/post-text-selection.gjs:229]: Unrelated?

File: lib/plugin/instance.rb [hunk 5]
```diff
@@ -1177,17 +1177,6 @@ def register_bookmarkable(klass)
     DiscoursePluginRegistry.register_bookmarkable(RegisteredBookmarkable.new(klass), self)
   end
 
-  ##
-  # Register an object that inherits from [Summarization::Base], which provides a way
-  # to summarize content. Staff can select which strategy to use
-  # through the `summarization_strategy` setting.
-  def register_summarization_strategy(strategy)
```
Comment (romanrizzi) [inline lib/plugin/instance.rb:1184]: Let's just deprecate the method for now and see if it's enough to make the build green. We can remove post-move

**Review by romanrizzi — CHANGES_REQUESTED** [review]

**Review by romanrizzi — APPROVED** [review]
